src/main/java/by/kukshinov/books/data/specefication/impl/sort/TitleSortSpecification.java

```diff
@@ -0,0 +1,14 @@
+package by.kukshinov.books.data.specefication.impl.sort;
+
+import by.kukshinov.books.data.specefication.SortBookSpecification;
+import by.kukshinov.books.model.Book;
+import by.kukshinov.books.model.comporators.TitleComparator;
+
+import java.util.Comparator;
+
+public class TitleSortSpecification extends SortBookSpecification {
```

Comment:
why do you need this herarchy? just create ComparatorFactory with switch(field), and use it in dao, and then call sort there